packages/headless/src/controllers/recent-queries-list/headless-recent-queries-list.ts

```diff
@@ -0,0 +1,148 @@
+import {ArrayValue, NumberValue, Schema} from '@coveo/bueno';
+import {buildController, Controller} from '../controller/headless-controller';
+import {search, recentQueries} from '../../app/reducers';
+import {loadReducerError} from '../../utils/errors';
+import {SearchEngine} from '../../app/search-engine/search-engine';
+import {RecentQueriesSection} from '../../state/state-sections';
+import {RecentQueriesState} from '../../features/recent-queries/recent-queries-state';
+import {
+  clearRecentQueries,
+  registerRecentQueries,
+} from '../../features/recent-queries/recent-queries-actions';
+import {
+  validateInitialState,
+  validateOptions,
+} from '../../utils/validate-payload';
+import {executeSearch} from '../../features/search/search-actions';
+import {updateQuery} from '../../features/query/query-actions';
+import {
+  logClearRecentQueries,
+  logRecentQueryClick,
+} from '../../features/recent-queries/recent-queries-analytics-actions';
+
+export interface RecentQueriesListProps {
+  /**
+   * The initial state that should be applied to the `RecentQueriesList` controller.
+   */
+  initialState?: RecentQueriesListInitialState;
+  /**
+   * The configuration options that should be applied to the `RecentQueriesList` controller.
+   */
+  options?: RecentQueriesListOptions;
+}
+
+export interface RecentQueriesListInitialState {
+  /**
+   * The list of recent queries
+   * @defaultValue `[]`
+   */
+  queries: string[];
+}
+
+const initialStateSchema = new Schema<RecentQueriesListInitialState>({
+  queries: new ArrayValue({required: false}),
+});
+
+export interface RecentQueriesListOptions {
+  /**
+   * The maximum number of queries to retain in the list.
+   * @defaultValue `10`
+   */
+  maxQueries: number;
```

Comment:
Could be `maximumLength`, or just `maximum` ? 

Since it's already inside a `RecentQueriesList` , the `queries` keyword is already implied/redundant.